### PR TITLE
Add image background component

### PR DESCRIPTION
**Pull request by gunnigylfa**
Added the <ImageBackground/> component which is intended for having nested components Inside of an Image. 
RN docs state that:

> A common feature request from developers familiar with the web is background-image. To handle this use case, you can use the  <ImageBackground> component, which has the same props as <Image>, and add whatever children to it you would like to layer on top of it.

There is a slight difference between props are I have added them to this. But other than that I just copied the ImageRe interface and implementation.

Edit: I reimplement these bindings so they share common props as much I could and got rid of the functor so there is far less duplicate code.

Here is an example of the component being used as a background image with a <Text/> as it's child.

image

### CODE REVIEW

File: src/components/imageBackgroundRe.re [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      | `auto => "auto"
+      | `resize => "resize"
+      | `scale => "scale"
+      };
+    let encodeDefaultSource (x: defaultSource) =>
+      switch x {
+      | URI x => rawImageSourceJS x
+      | Required x => rawImageSourceJS x
+      };
+    let make
```
Comment (wokalski) [inline src/components/imageBackgroundRe.re]: This module should only contain `make`.

We don't need a functor unless there exists `Animated.ImageBackground`. The other types and utility functions should be the same as in `ImageRe.re`. If they are unaccessible due to `reis` feel free to put them in a separate file. Like `ImageCommonRe.re`.

**Comment by wokalski** [comment]
Ah. sorry for confusion. The `ImageRe` functor won't work because of different props. We need to keep your code, however it'd be great to eliminate the duplication.

**Comment by gunnigylfa** [comment]
@wokalski  Yeah I just realized that when trying to fix the RNTester example once I changing it.  I might try to extract the two interfaces two a common one once I have gotten more familiar with the workings of reason. Would this implementation be sufficient for now?

**Comment by arnarthor** [comment]
I agree that less duplication would be great, but should we merge this for now and try to optimize later. Keeping this open too long might make everyone lose context when coming back to this.

**Comment by wokalski** [comment]
And merging it will introduce duplication and simply wrong code. The functor is not necessary here and the types which are the same as in Image should simply reference the types in Image. Otherwise the types become different and you cannot pass the same image from `Image` to `ImageBackground` because types are nominal

**Comment by arnarthor** [comment]
Riiight, I misunderstood your comment. Thought you had taken the review back with the second comment.

I agree, that all makes sense 👍. @gunnigylfa I can help if needed as well

**Comment by gunnigylfa** [comment]
Ah okay I thought that you meant keep the code as it was but thanks for the clarification @wokalski  I will try and fix this as soon as I can, I will probably take you up on that offer @arnarthor  😄

**Comment by gunnigylfa** [comment]
@wokalski @arnarthor I managed to get this working again with a different implementations of the bindings. Now I'm not using the functor and I'm sharing types with `ImageRe` so an Image can be sent into a ImageBackground just as it could be sent into an Image.

**Comment by wokalski** [comment]
Awesome. It looks good. Sorry for nitpicking!

**Comment by gunnigylfa** [comment]
No worries @wokalski I appreciate the feedback 🙂

**Pull request merged**